core/src/components/modal/modal.spec.ts

```diff
@@ -68,10 +68,13 @@ describe('modal', () => {
 		});
 		const directive = modal.directives.modalDirective(element);
 		const promise = modal.api.open();
-		modal.actions.closeButtonClick({} as any as MouseEvent);
+		const closeButton = document.createElement('button');
+		const closeDirective = modal.directives.closeButtonDirective(closeButton);
+		closeButton.click();
```

Comment:
```suggestion
attachDirectiveAndClick(modal.directives.closeButtonDirective)
```
Didn't you create this helper function exactly for that purpose?